wiki/configuration/getting-started/en.md

```diff
@@ -0,0 +1,153 @@
+
+Something incredibly important to remember is that you can't just send any class into `TrackedValue`'s generics: Quilt Config has to know how to serialize it. By default, all [primitive types](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html) are serializable, as well as the `String` class. We'll cover how to explain to Quilt Config how to serialize a class later, but first: more fun with annotations!
+
+Anyway. New value!
+
+`src/main/com/example/example_mod/ExampleModConfig`:
+
+```java
+public class ExampleModConfig extends ReflectiveConfig {
+	// ...
+	@Comment("Whether to print the message at all.")
+	public final TrackedValue<Boolean> print = this.value(true);
+}
+```
+
+Here we simply leverage the `Comment` annotation to add a comment explaining exactly what the config field does. This will be saved by Quilt Config in the config file, as long as the format you use supports comments. Neat!
+
+Finally, we're going to use one more annotation to tie it all together. By default, Quilt Config serializes to [TOML](https://toml.io/en/) which has a couple standards we're not following. In Java, `public final` fields are named using `lowerCamelCase`, like our `timesToPrint` field. But, in a TOML file, names should use `snake_case`, meaning we should have named the field `times_to_print`. To match both conventions, we can use the `SerializedName` annotation!
+
+`src/main/com/example/example_mod/ExampleModConfig`:
+
+```java
+public class ExampleModConfig extends ReflectiveConfig {
+	// ...
+	@IntegerRange(min = 0, max = 10)
+    @SerializedName("times_to_print")
+	public final TrackedValue<Integer> timesToPrint = this.value(1);
+	// ...
+}
+```
+
+Problem solved! Now it'll follow the Java conventions in the Java code, and the TOML conventions in the saved TOML file. We can apply `SerializedName` annotations to every multi-word config field to solve our problems everywhere.
+
+## Maps and lists
+
+Two of the most common data structures used in Java programming are `Map` and `List`. Quilt Config provides convenient serializable versions of both of these in the for of `ValueMap` and `ValueList`!
+
+Starting with maps, `ReflectiveConfig` provides us a handy `map` method to help us easily make a default value.
+
+`src/main/com/example/example_mod/ExampleModConfig`:
+
+```java
+public class ExampleModConfig extends ReflectiveConfig {
+	// ...
+	@Comment("When a key in this map shows up in the message, it'll be replaced with the corresponding value.")
+	public final TrackedValue<ValueMap<String>> replacements = this.map("")
+		.put("oro armor", "rai minecraft")
+		.put("lambda aurora", "rai minecraft")
+		.put("tib s", "rai minecraft")
+		.build();
+}
+```
+
+There are a few things of note here:
+- We have to pass an empty string (`""`) to the `map` method, in order for it to understand that the map stores values of the type `String`. This value is not used beyond checking the type, so you can put whatever you'd like there.
```

Comment:
The first argument is [the default value](https://github.com/QuiltMC/quilt-config/blob/5c016b4042af6f68c198935017f49c6d18ab8647/src/main/java/org/quiltmc/config/api/ReflectiveConfig.java#L111) btw

Reply:
not for maps, I have it on glitch's authority that this is nonsense because I was super confused and pinged him about it